plugins/mongodb/mongodbembed.cpp

```diff
@@ -229,17 +213,20 @@ namespace mongodbembed
                 end++;
 
             result += std::string(start, end++ - start); // Only add the data inside the quotation marks to result string
-            while (*end && *end != '}') 
-                end++; // Only have to get out of one nested document
-            end++;
-            depth--;
-            row = end;
-            start = end;
         }
         else
         {
             failx("EJSON datatype error: '%s' is not supported in the current version.", key.c_str());
         }
+
+        // Get out of nested object.
+        while (*end && *end != '}')
+            end++;
+
+        end++;
```

Comment:
if *end is '\0' (which I imagine should never happen) then this will increment `end` to point past the end of the string.

Reply:
I added a check for the null character in case there is invalid json or an error while processing.